README.md

```diff
@@ -36,6 +36,62 @@ Cons: Only includes standard libraries - no Debian `-dev` packages.
 
 [MentorGraphics toolchain](http://sourcery.mentor.com/public/gnu_toolchain/arm-none-linux-gnueabi/arm-2014.05-29-arm-none-linux-gnueabi.exe) (formerly known as CodeSourcery).
 
+### Windows 10
+
+Windows 10 supports the Ubuntu Subsystem for Windows which allows us to install and execute the entire compiler toolchain. The steps required to compile native-mode EV3 applications on Windows 10 is as follows:
+
+1. Be sure that your Windows 10 installation has Windows Subsystem for Linux installed. To install it, go to Control Panel --> Programs and Features --> Turn Windows Feature On/Off and check the box next to Windows Subsystem for Linux (Beta).
+
+2. Fire up the bash shell by pressing Start Key, type `bash` and press `Enter`. This will open up Bash on Ubuntu on Windows.
+
+3. Install the ARM compiler by typing `sudo apt-get install arm-linux-gnueabi-gcc`.
+
+4. Use `git clone` to clone this repository to the directory of your choice, e.g.,
+
+```
+git clone https://github.com/ddemidov/ev3dev-lang-cpp.git
+```
+
+will clone the repo into a directory called `/ev3dev-lang-cpp.git`.
+
+5. You need to make some changes to the top-level `CMakeLists.txt` file. First, go into the directory
+
+```
+cd ev3dev-lang-cpp
+```
+Now, edit the `CMakeLists.txt` file with a text editor of your choice, e.g.,
+
+```
+vi CMakeLists.txt
+```
+
+Just after the `project(...)` declaration, set the C/C++ compilers by adding the following lines:
+
+```
+set(CMAKE_CC_COMPILER "arm-linux-gnueabi-gcc")
+set(CMAKE_CC_COMPILER "arm-linux-gnueabi-g++")
+```
+
+6. You also need the following define, without which the nanosleep-using demos won't compile:
+
+```
+add_definitions(-D_GLIBCXX_USE_NANOSLEEP)
```

Comment:
We should probably just update our `CMakeLists.txt` to do this by default. [g++ does not need this since 4.8](http://stackoverflow.com/questions/12523122/what-is-glibcxx-use-nanosleep-all-about), but this is not the first issue I see where this is needed for crosscompiler.